### ISSUE TITLE
pending should be cleaned while calling auction_failed/auction_succeeded

### ISSUE THREAD

**Issue by GopherJ** [parallel-finance/parallel#1052]

**Comment by GopherJ** [comment]
or maybe make it possible to claim pending contribution

**Comment by GopherJ** [comment]
@drbh can you take this. We should have a succeed phase between `Closed` and `Expired` so that the frontend can know this parachain won the auction.

`slot_expired` will then need to check:

```
ensure!(vault.phase == VaultPhase::Expired, Error::<T>::IncorrectPhase);
```

this change will be included in `v1.7.5-rc1` together with @ashi20192020ayush 's PR

**Comment by GopherJ** [comment]
currently `migrate_pending` must be called before entering `Contributing` phase and we dont allow to call migrate_pending in other phase so this can be closed.

`migrate_pending` shouldn't be called in `Contributing` phase because there maybe
